rustls/src/client/mod.rs

```diff
@@ -161,8 +161,16 @@ impl ClientConfig {
         // is PACKET_OVERHEAD.
         if let Some(x) = *mtu {
             use crate::msgs::fragmenter;
-            debug_assert!(x > fragmenter::PACKET_OVERHEAD);
-            self.mtu = Some(x - fragmenter::PACKET_OVERHEAD);
+            if !cfg!(test) {
+                debug_assert!(x > fragmenter::PACKET_OVERHEAD);
+            }
```

Comment:
Seems like `!cfg!(test)` is the wrong way around for this? We want to assert only when in `cfg!(test)`, right?

Also I'd maybe move that `if` into the `self.mtu.is_none()` branch? Makes it a little clearer IMO.

Reply:
> We want to assert only when in cfg!(test), right?

I don't think so. If done that way, the test will fail without being able to check the resulting value.

> Also I'd maybe move that if into the self.mtu.is_none() branch? Makes it a little clearer IMO.

Will do.